semantics/src/main/scala/aqua/semantics/expr/func/ForSem.scala

```diff
@@ -75,18 +78,39 @@ class ForSem[S[_]](val expr: ForExpr[S]) extends AnyVal {
 
 object ForSem {
 
+  def itemOrPair[S[_]](nameOrPair: ForExpr.NameOrPair[S]): (String, Option[ForKeyValue]) =
+    nameOrPair match {
+      case Right(v) => (v.value, None)
+      case Left(k, v) => ("-iterable-", ForKeyValue(k.value, v.value).some)
+    }
+
   def beforeFor[S[_], F[_]: Monad](
-    item: Name[S],
+    item: ForExpr.NameOrPair[S],
     iterable: ValueToken[S]
   )(using
     V: ValuesAlgebra[S, F],
     N: NamesAlgebra[S, F],
-    T: TypesAlgebra[S, F]
+    M: ManglerAlgebra[F]
   ): F[Option[ValueRaw]] = (for {
     value <- V.valueToIterable(iterable)
     (raw, typ) = value
-    _ <- OptionT.liftF(
-      N.define(item, typ.element)
-    )
-  } yield raw).value
+    res <- (typ, item) match {
+      case (smt: StreamMapType, Left(key, value)) =>
+        OptionT.liftF(for {
+          _ <- N.define(key, ScalarType.string)
+          _ <- N.define(value, smt.element)
+        } yield raw)
+      case (smt: StreamMapType, Right(it)) =>
+        val typeName = "-streamMapIter-"
+        OptionT.liftF(for {
+          newTypeName <- M.rename(typeName)
```

Comment:
Do we need a unique type name?

Reply:
I believe it is better to give this type a static name like `KVPair`. It will be shown in errors, e.g.
```aqua
aqua Main

export main

data Test:
  key: string
  valueA: u64

func test(t: Test) -> u64:
  <- t.valueA

func main():
  map: %u64
  for it <- map:
    test(it)
```
returns
```
---- Error: /home/vagrant/workspace/aqua/aqua-src/antithesis.aqua:15:10
15     test(it)
            ^^=
     Wrong value type, expected: data Test{key: string, valueA: u64}, given: data -streamMapIter--0{key: string, value: u64}
```
Which is not very pretty

Reply:
static name can break compilation on using multiple maps with multiple fors. But it can be prettier still